railties/test/application/rake/multi_dbs_test.rb

```diff
@@ -195,6 +238,45 @@ def db_migrate_name_dumps_the_schema(name, schema_format)
         end
       end
 
+      def db_migrate_env_dumps_the_schema(name, schema_format)
+        add_to_config "config.active_record.schema_format = :#{schema_format}"
+        require "#{app_path}/config/environment"
+
+        Dir.chdir(app_path) do
+          generate_models_for_animals
+
+          assert_not(File.exist?("db/schema.rb"))
+          assert_not(File.exist?("db/animals_schema.rb"))
+          assert_not(File.exist?("db/structure.sql"))
+          assert_not(File.exist?("db/animals_structure.sql"))
+
+          ENV["DATABASE_NAME"] = name
```

Comment:
This env should be set and unset in the tests below instead of in the method definition. Otherwise it will be always set for tests after this.